spec/projection_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'check_graphite/projection'
+require 'nagios_check'
+
+describe describe CheckGraphite::Projection do
+  let :check do
+    Class.new do
+      attr_reader :values
+      include NagiosCheck
+      include CheckGraphite::Projection
+    end.new
+  end
+
+  describe 'primary value' do
+    subject do
+      check.prepare
+      check.send(:parse_options, ['--projection', projection])
+      check.options.processor.call(datapoints)
+      check.values.first[1]
+    end
+
+    context 'given a linearly decreasing series and a projection of 2 sec' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [9,1], [8,2]] }
+
+      it { should be_within(0.01).of(6) }
+    end
+
+    context 'given a constant series and a projection of 2 sec' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [10,1], [10,2]] }
+
+      it { should be_within(0.01).of(10) }
+    end
+  end
+
+  describe 'p-value' do
+    subject do
+      check.prepare
+      check.send(:parse_options, ['--projection', projection])
+      check.options.processor.call(datapoints)
+      check.values['p-value']
+    end
+
+    context 'given a constant series (where Pearson is undefined)' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [10,1], [10,2]] }
+      it { should eq('undefined') }
```

Comment:
This is an example where the description-less syntax kind of works: 'CheckGraphite::Projection p-value given a constant series (where Pearson is undefined) should eq "undefined"'. However, adding a hand-crafted description would be even better: "… given a constant series (where Pearson is undefined) returns undefined", but it's marginal.